src/main/java/teammates/common/util/Config.java

```diff
@@ -95,7 +95,7 @@ private String getGcsBucketname() {
     }
 
     private String getAppUrl() {
-        return props.getProperty("app.url");
+        return Url.trimTrailingSlash(props.getProperty("app.url"));
     }
```

Comment:
I suggest the change because that will make this the factory method we've been talking about.